pkg/controllers/jobset_controller.go

```diff
@@ -110,8 +114,26 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	// If JobSet is already completed or failed, clean up active child jobs.
+	// If JobSet is already completed or failed, clean up active child jobs and requeue if TTLSecondsAfterFinished is set.
 	if jobSetFinished(&js) {
+		if js.Spec.TTLSecondsAfterFinished != nil {
+			expired, err := r.checkIfTTLExpired(ctx, &js)
+			if err != nil {
+				log.Error(err, "checking if TTL expired")
+				return ctrl.Result{}, err
+			}
+			// if expired is true, that means the TTL has expired, and we should delete the JobSet
+			// otherwise, we requeue it for the remaining TTL duration.
+			if expired {
+				log.V(5).Info("JobSet TTL expired")
+				if err := r.deleteJobSet(ctx, &js); err != nil {
+					log.Error(err, "deleting jobset")
+					return ctrl.Result{}, err
+				}
+			} else {
+				return ctrl.Result{RequeueAfter: requeueJobSetAfter(&js)}, nil
+			}
```

Comment:
can we please more this logic to a function named `executeTTLAfterFinishedPolicy` similar to the pattern we have for failure and success policies below?

Reply:
+1